tests/unit/backend/recipe/algorithm/test_MantidSnapper.py

```diff
@@ -18,7 +20,7 @@ def test_snapper_fake_algo():
         return_of_algo = "return of algo"
         mantidSnapper = MantidSnapper(parentAlgorithm=None, name="")
         test = mantidSnapper.fakeFunction("test", fakeOutput=return_of_algo)
-        assert test == return_of_algo
+        assert str(test.__class__) == str(callback(return_of_algo.__class__).__class__)
```

Comment:
I think this test is less stringent than the old one.  The old one ensured the value was the same, whereas this is apparently checking that a callback is returned.

Reply:
Thats because an unpopulated callback is now returned instead.  This is mostly so all types are treated uniformly as opposed to workspace names being an exception. 

In order to actually test the value, the algo would have to run and populate that callback.